datahub-web-react/src/app/entity/shared/tabs/Dataset/Validations/utils.tsx

```diff
@@ -0,0 +1,324 @@
+    };
+};
+
+// /**
+//  * Returns a list of assertion groups, where assertions are grouped
+//  * by their "type" or "category". Each group includes the assertions inside, along with
+//  * a summary of passing and failing assertions for the group.
+//  *
+//  * @param assertions The assertions to group
+//  */
+export const createAssertionGroups = (assertions: Array<Assertion>): AssertionGroup[] => {
+    // Pre-sort the list of assertions based on which has been most recently executed.
+    assertions.sort(sortAssertions);
+
+    const typeToAssertions = new Map();
+    assertions
+        .filter((assertion) => assertion.info?.type)
+        .forEach((assertion) => {
+            const groupType = assertion.info?.type;
+            const groupedAssertions = typeToAssertions.get(groupType) || [];
+            groupedAssertions.push(assertion);
+            typeToAssertions.set(groupType, groupedAssertions);
+        });
+
+    // Now, create summary for each type and build the AssertionGroup object
+    const assertionGroups: AssertionGroup[] = [];
+    typeToAssertions.forEach((groupedAssertions, type) => {
+        const newGroup: AssertionGroup = {
+            name: getAssertionGroupName(type),
+            icon: getAssertionGroupTypeIcon(type),
+            assertions: groupedAssertions,
+            summary: getAssertionsSummary(groupedAssertions),
+            type,
+        };
+        assertionGroups.push(newGroup);
+    });
+
+    return assertionGroups;
+};
+
+// TODO: Make this the default inside DatasetAssertionsSummary.tsx.
+export const getAssertionGroupSummaryIcon = (summary: AssertionStatusSummary) => {
+    if (summary.total === 0) {
+        return null;
+    }
+    if (summary.passing === summary.total) {
+        return <StyledCheckOutlined />;
+    }
+    if (summary.erroring > 0) {
+        return <StyledExclamationOutlined />;
+    }
+    return <StyledCloseOutlined />;
+};
+
+// TODO: Make this the default inside DatasetAssertionsSummary.tsx.
+export const getAssertionGroupSummaryMessage = (summary: AssertionStatusSummary) => {
+    if (summary.total === 0) {
+        return 'No assertions have run';
+    }
+    if (summary.passing === summary.total) {
+        return 'All assertions are passing';
+    }
+    if (summary.erroring > 0) {
+        return 'An error is preventing some assertions from running';
+    }
+    if (summary.failing === summary.total) {
+        return 'All assertions are failing';
+    }
+    return 'Some assertions are failing';
+};
+
+export const getAssertionTypesForEntityType = (entityType: EntityType, monitorsConnectionForEntityExists: boolean) => {
+    return ASSERTION_INFO.filter((type) => type.entityTypes.includes(entityType)).map((type) => ({
+        ...type,
+        enabled: type.enabled && (!type.requiresConnectionSupportedByMonitors || monitorsConnectionForEntityExists),
+    }));
+};
+
+// export const getEntityUrnForAssertion = (assertion: Assertion) => {
```

Comment:
we should revert this comment

Reply:
I am removing this method as we are not using this in OSS